mlflow/server/js/src/gateway/api.ts

```diff
@@ -28,8 +28,27 @@ import type {
   CreateEndpointBindingRequest,
   CreateEndpointBindingResponse,
   ListEndpointBindingsResponse,
+  SecretInfo,
 } from './types';
 
+// Raw API response type before transformation (has auth_config_json string instead of auth_config object)
+interface RawSecretInfo extends Omit<SecretInfo, 'auth_config'> {
+  auth_config_json?: string;
```

Comment:
I may forget past conversation, we still serialize `auth_config_json` since it't not always `Map<string, string?`?